assembl/static2/js/app/components/common/sideMenuTree.jsx

```diff
@@ -0,0 +1,110 @@
+// @flow
+import React from 'react';
+import { Link } from 'react-router';
+import classNames from 'classnames';
+import { getPartialTree, getChildren } from '../../utils/tree';
+import { SECTION_INDEX_GENERATOR } from '../../utils/section';
+
+type SideMenuTreeState = {
+  activeKey: ?string,
+  show: boolean
+};
+
+type SideMenuTreeProps = {
+  rootIdea: Object,
+  synthesisPostId: string,
+  subIdeas: Array<Object>,
+  index: number,
+  slug: string,
+  parents: Array<number>,
+  indexGenerator: Function
+};
+
+class SideMenuTree extends React.Component<*, SideMenuTreeProps, SideMenuTreeState> {
+  props: SideMenuTreeProps;
+
+  state: SideMenuTreeState;
+
+  static defaultProps = {
+    indexGenerator: SECTION_INDEX_GENERATOR.alphanumericOr
+  };
+
+  constructor(props: SideMenuTreeProps) {
+    super(props);
+    this.state = {
+      activeKey: null,
+      show: false
+    };
+  }
+
+  getLinkContainerClassNames = (id: string) => (location.hash === `#${id}` ? 'link-container active' : 'link-container');
+
+  getIndexes = () => {
+    const { index, parents } = this.props;
+    const indexes = parents.slice();
```

Comment:
I don't understand this `.slice()`. Isn't it the same as `const indexes = parents` ?

Reply:
Correct. I copied this function from section.jsx since I wanted to achieve the same purpose, I can correct this line and extract this function to utils

Reply:
Moved the function to utils/section.js
Replaced parents.slice() by [...parents]

Reply:
Please don't replace parents.slice() by [...parents]. `indexes` is a copy of `parents`...

Reply:
ok